zerocopy-derive/src/lib.rs

```diff
@@ -401,6 +399,147 @@ const STRUCT_UNION_ALLOWED_REPR_COMBINATIONS: &[&[StructRepr]] = &[
     &[StructRepr::C, StructRepr::Packed],
 ];
 
+fn derive_try_from_bytes_enum(ast: &DeriveInput, enm: &DataEnum) -> proc_macro2::TokenStream {
+    if !enm.is_fieldless() {
+        return Error::new_spanned(ast, "only field-less enums can implement TryFromBytes")
+            .to_compile_error();
+    }
+
+    let reprs = try_or_print!(ENUM_TRY_FROM_BYTES_CFG.validate_reprs(ast));
+    let discriminant_type = match reprs.as_slice() {
```

Comment:
Unsoundness: Any of these types can be shadowed, so you'll need to refer to them instead from `core_reexport::primitive`.

You should also add an integration test to make sure we don't regress on this.

Reply:
Done on both counts